### DISCUSSION TITLE
Associating a display name after an OAuth sign up

### DISCUSSION THREAD

**Discussion by jonohewitt** [supabase/supabase#2238] [Questions] [answered]
Hey everyone, I'm trying to set up a create account form where a user can enter a display name, check that it's unique, then choose to create an account with either an OAuth provider or email  / password. 

The issue I'm having with the OAuth option is that I can't associate a display name to a user before they've completed the third party authentication, but the return from the external provider seems to be a fresh page load. Does anyone have any advice for how to automatically append this info after account creation? Thanks!

**Comment by silentworks** [comment, marked as answer]
I do this by checking if the `displayName` column of my `public.users` (created by me) table is empty and then display a modal window with a form field to enter display name.

**Reply by jonohewitt** [reply]
Thanks @silentworks, that sounds like a reasonable work around - it feels a little weird doing it in two steps and having a limbo state without a display name if the user navigates away before submitting the display name - but I'll give it a go!
